### PR TITLE
Remove K8sComponentsByVersionMap from AgentBaker.

### PR DESCRIPTION
**Pull request by yizhang4321**
AgentBaker now requires user/caller to pass in K8sComponentsByVersionMap. This removes the need to keep AgentBaker's K8sComponentsByVersionMap in sync with its user/caller.

### CODE REVIEW

File: pkg/agent/types.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ type KeyVaultRef struct {
 type NodeBootstrappingConfiguration struct {
 	ContainerService              *datamodel.ContainerService
 	CloudSpecConfig               *datamodel.AzureEnvironmentSpecConfig
+	K8sComponentsByVersionMap     map[string]map[string]string
```
Comment (xuto2) [inline pkg/agent/types.go:23]: This is a bit overkill as it contains many unnecessary data, it's also not specific enough for caller. There isn't  a lot of values AgentBaker need to get from this list, can we pass those needed properties explicitly, either through the ContainerService object or as a new property?

Comment (yizhang4321) [reply pkg/agent/types.go]: I changed the code to pass in just the K8sComponents, not the entire K8sComponentsByVersionMap.

We can also change the code to pass in individual values used, but that will make the NodeBootstrappingConfiguration struct much bigger. I know of at least 3 values we. use from K8sComponents: the "pause" image path, the "hyperkube" image path, and "windowszip".

File: pkg/agent/types.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ type KeyVaultRef struct {
 type NodeBootstrappingConfiguration struct {
 	ContainerService              *datamodel.ContainerService
 	CloudSpecConfig               *datamodel.AzureEnvironmentSpecConfig
+	K8sComponents                 map[string]string
```
Comment (xuto2) [inline pkg/agent/types.go:23]: we only need a few value from this, hyperkube, pause,winzip and nodestatusfreq. Can we make a struct to hold them? I feel that's more explicit than a map

Comment (yizhang4321) [reply pkg/agent/types.go]: Code updated to take in a struct with explicit listing of the strings we use.

File: pkg/agent/params.go [hunk 3]
```diff
@@ -176,11 +175,7 @@ func assignKubernetesParameters(properties *datamodel.Properties, parametersMap
 				addValue(parametersMap, "kubeBinaryURL", kubernetesConfig.CustomKubeBinaryURL)
 			}
 
-			kubernetesHyperkubeSpec := hyperkubeImageBase + k8sComponents["hyperkube"]
-			if kubernetesConfig.CustomHyperkubeImage != "" {
-				kubernetesHyperkubeSpec = kubernetesConfig.CustomHyperkubeImage
-			}
```
Comment (xuto2) [inline pkg/agent/params.go:182]: reminder: don't forget to move this logic to RP side

File: pkg/agent/params.go [hunk 4]
```diff
@@ -230,13 +225,7 @@ func assignKubernetesParameters(properties *datamodel.Properties, parametersMap
 			addValue(parametersMap, "enableAggregatedAPIs", kubernetesConfig.EnableAggregatedAPIs)
 
 			if properties.HasWindows() {
-				// Kubernetes packages as zip file as created by scripts/build-windows-k8s.sh
-				// will be removed in future release as if gets phased out (https://github.com/Azure/aks-engine/issues/3851)
-				kubeBinariesSASURL := kubernetesConfig.CustomWindowsPackageURL
-				if kubeBinariesSASURL == "" {
-					kubeBinariesSASURL = cloudSpecConfig.KubernetesSpecConfig.KubeBinariesSASURLBase + k8sComponents["windowszip"]
-				}
```
Comment (xuto2) [inline pkg/agent/params.go:238]: same reminder here.

File: pkg/agent/params.go (hunk 3 above)
Comment (yizhang4321) [reply pkg/agent/params.go:182]: Yes, next time we vendor in AgentBaker, I'll go into that branch apply the same logic to generate the strings.

File: pkg/agent/params.go (hunk 4 above)
Comment (yizhang4321) [reply pkg/agent/params.go:238]: Acknowledged.

File: pkg/agent/baker.go [hunk 5]
```diff
@@ -395,7 +395,7 @@ func getContainerServiceFuncMap(config *NodeBootstrappingConfiguration) template
 			return datamodel.AzureADIdentitySystem
 		},
 		"GetPodInfraContainerSpec": func() string {
-			return cs.Properties.OrchestratorProfile.GetPodInfraContainerSpec()
+			return config.K8sComponents.PodInfraContainerImageURL
```
Comment (xuto2) [inline pkg/agent/baker.go:398]: @eriksywu FYI

**Pull request merged**